builder.py

```diff
@@ -0,0 +1,54 @@
+# Copyright (c) 2023, elParaguayo. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+import sys
+
+from setuptools import build_meta as _orig
+from setuptools.build_meta import *  # noqa: F401,F403
+
+WAYLAND_DEPENDENCIES = ["pywlroots>=0.16.4,<0.17.0"]
+
+
+def wants_wayland(config_settings):
+    if config_settings:
+        for key in ["Backend", "backend"]:
+            if config_settings.get(key, "") in ("Wayland", "wayland"):
```

Comment:
config_settings.get(key, "").lower() == "wayland"

Reply:
same for backend maybe?

Reply:
or maybe we should just make the key --wayland?

Reply:
Not possible. There are different builders (`pip`, `python -m build` etc) so the only standard way to pass arguments is the config-settings as that's part of the spec.

Reply:
> or: config_settings.get(key, "").lower() == "wayland"

Ha. I thought about this and went with the other option.

Reply:
> Not possible. There are different builders (`pip`, `python -m build` etc) so the only standard way to pass arguments is the config-settings as that's part of the spec.

ah yeah I meant --config-setting wayland or --config-setting --wayland (but that would probably not work). Let's leave it like this and maybe do the lower thing but that's optional